parser_test.go

```diff
@@ -16,25 +16,32 @@ import (
 var errKeyFuncError error = fmt.Errorf("error loading key")
 
 var (
-	jwtTestDefaultKey      *rsa.PublicKey
-	jwtTestRSAPrivateKey   *rsa.PrivateKey
-	jwtTestEC256PublicKey  crypto.PublicKey
-	jwtTestEC256PrivateKey crypto.PrivateKey
-	defaultKeyFunc         jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return jwtTestDefaultKey, nil }
-	ecdsaKeyFunc           jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return jwtTestEC256PublicKey, nil }
-	emptyKeyFunc           jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return nil, nil }
-	errorKeyFunc           jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return nil, errKeyFuncError }
-	nilKeyFunc             jwt.Keyfunc = nil
+	jwtTestDefaultKey       *rsa.PublicKey
+	jwtTestRSAPrivateKey    *rsa.PrivateKey
+	jwtTestEC256PublicKey   crypto.PublicKey
+	jwtTestEC256PrivateKey  crypto.PrivateKey
+	exampleCognitoPublicKey crypto.PublicKey
+	defaultKeyFunc          jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return jwtTestDefaultKey, nil }
+	ecdsaKeyFunc            jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return jwtTestEC256PublicKey, nil }
+	exampleCognitoKeyFunc   jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return exampleCognitoPublicKey, nil }
+	emptyKeyFunc            jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return nil, nil }
+	errorKeyFunc            jwt.Keyfunc = func(t *jwt.Token) (interface{}, error) { return nil, errKeyFuncError }
+	nilKeyFunc              jwt.Keyfunc = nil
 )
 
 func init() {
 	// Load public keys
 	jwtTestDefaultKey = test.LoadRSAPublicKeyFromDisk("test/sample_key.pub")
 	jwtTestEC256PublicKey = test.LoadECPublicKeyFromDisk("test/ec256-public.pem")
 
+	// Load cognito public key - note there is only a public key for this key pair and should only be used for the
+	// two test cases below.
+	exampleCognitoPublicKey = test.LoadECPublicKeyFromDisk("test/exampleCognito-public.pem")
```

Comment:
I would suggest avoiding a product name here and just name it paddedKey or something

Reply:
done